### PR TITLE
save-demo-agent-distribution

### PR DESCRIPTION
**Pull request by sanyavertolet**
### What's done:
 * Added `agent.toml` and `save-demo-agent.kexe` to `save-demo-agent-distribution.jar`
 * Implemented simple config parsing logic for `save-demo-agent`
 * Fixed workflow file in order to upload `save-demo-agent-distribution.jar`

### CODE REVIEW

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/utils/FileUtils.kt [hunk 1]
```diff
@@ -43,3 +44,16 @@ expect inline fun <reified C : Any> parseConfig(configPath: Path): C
...
+ * Notice that [C] should be serializable
+ *
+ * @param defaultConfig config that should be set if no config was found ([FileNotFoundException])
+ * @param configName name of a toml config file, agent.toml by default
+ * @return [C] filled with configuration information
+ */
+inline fun <reified C : Any> parseConfigOrDefault(
+    defaultConfig: C,
+    configName: String = "agent.toml",
+): C = try { parseConfig(configName) } catch (e: FileNotFoundException) { defaultConfig }
```
Comment (0x6675636b796f75676974687562) [inline save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/utils/FileUtils.kt:59]: I would still log a message that 

> Config file %s not found, falling back to the default config...

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/utils/FileUtils.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/utils/FileUtils.kt:59]: Sure, will add

File: save-demo-agent/build.gradle.kts [hunk 2]
```diff
@@ -58,4 +59,18 @@ kotlin {
             }
         }
     }
+
+    @Suppress("GENERIC_VARIABLE_WRONG_DECLARATION")
```
Comment (kgevorkyan) [inline save-demo-agent/build.gradle.kts:63]: still need this suppress? you already provide a type `TaskProvider<KotlinNativeLink>`?

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/utils/FileUtils.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanyavertolet) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/utils/FileUtils.kt:59]: Oh ok

File: save-demo-agent/build.gradle.kts (hunk 2 above)
Comment (sanyavertolet) [reply save-demo-agent/build.gradle.kts:63]: As I understand, due to kts features, we still need to pass a type to `tasks.named`, but diktat fixes it into 
```kotlin
val linkTask: TaskProvider<KotlinNativeLink> = tasks.named("linkReleaseExecutableLinuxX64")
```
which is wrong.

Comment (sanyavertolet) [reply save-demo-agent/build.gradle.kts:63]: So yes, we still need it

File: save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/utils/FileUtils.kt (hunk 1 above)
Comment (sanyavertolet) [reply save-cloud-common/src/commonMain/kotlin/com/saveourtool/save/utils/FileUtils.kt]: Done

**Review by nulls — APPROVED** [review]

**Pull request merged**